External/FEXCore/Source/Interface/Core/JIT/x86_64/BranchOps.cpp

```diff
@@ -80,8 +80,8 @@ DEF_OP(ExitFunction) {
     Label l_BranchHost;
     Label l_BranchGuest;
 
-    lea(rax, ptr[rip + l_BranchHost]);
```

Comment:
rsi is the second function parameter register. So switching the assignment to rsi fills the record parameter in
void X86ExitFunctionLinkerCodeAsm( FEXCore::Core::CpuStateFrame *FillMe, uint64_t *record )

I did not recall that the original code saved and restored it, and the calls to the function pointers may trash it, so I assume it is otherwise unassigned at this point.

Reply:
Ah okay. A comment here explaining this would be good then

Reply:
Comment added in subsequent commit